src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -1531,7 +1531,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
                 /* check all base classes, since derived value classes might lurk in refinement parents */
                 if (acc.tpe.typeSymbol.baseClasses exists (_.isDerivedValueClass))
                   context.error(acc.pos, "value class may not wrap another user-defined value class")
-                checkEphemeral(clazz, body filterNot (stat => stat.symbol != null && stat.symbol.accessedOrSelf == paramAccessor))
+                checkEphemeral(clazz, body.filter(stat => stat.symbol == null || stat.symbol.accessedOrSelf != paramAccessor))
```

Comment:
`filterNot` doesn't delegate to `filter`, both call `filterCommon`. In this case, it's easier to read the "guard clause" `stat.symbol != null` as a short-circuit. If you name the condition `def referencesUnderlying` then it's very readable `body.filterNot(referencesUnderlying)`. Approximately,
```
def referencesUnderlying(sym: Symbol) = sym != null && sym.accessedOrSelf == paramAccessor
decls.toList.find(x => x.isMethod && referencesUnderlying(x)) match {
  case Some(accessor) =>
    if (accessor.tpe.typeSymbol.baseClasses.exists(_.isDerivedValueClass))
      context.error(accessor.pos, "value class may not wrap another user-defined value class")
    checkEphemeral(clazz, body.filterNot(stat => referencesUnderlying(stat.symbol)))
}
```

Reply:
I can do that. I found the double negation very confusing although it's quite common in my native tongue.